File: admin-ui/src/app/dtos/scene/ChangeSceneDTO.ts
import {SceneLightConfigDTO} from "./SceneLightConfigDTO";

export type ChangeSceneDTO = {
  id: number,
  name: string,
  defaultScene: boolean,
  spotifyResource: string | undefined,
  spotifyVolume: number,
  lights: SceneLightConfigDTO[],
};

File: backend-service/src/main/java/ch/micha/automation/room/scene/dtos/ChangeSceneDTO.java
package ch.micha.automation.room.scene.dtos;

import lombok.*;

import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@ToString
public class ChangeSceneDTO {
    private int id;
    private String name;
    private boolean defaultScene;
    private String spotifyResource;
    private int spotifyVolume;
    private List<SceneLightConfigDTO> lights;
}
